lib/libc/stdlib/lib_mktemp.c

```diff
@@ -71,26 +61,65 @@
  ****************************************************************************/
 
 /****************************************************************************
- * Public Functions
+ * Public Function Prototypes
  ****************************************************************************/
 
 /****************************************************************************
  * Name: mktemp
  *
  * Description:
  *   The use of mktemp is dangerous; use mkstemp instead.
+ *   The mktemp() function generates a unique temporary filename from template.
+ *   The last six characters of template must be XXXXXX and these are replaced
+ *   with a  string  that  makes the filename unique.
+ *   Since it will be modified, template must not be a string constant,
+ *   but should be declared as a character array.
+ *
+ * Return Value:
+ *   The mktemp() function always returns template.
+ *   If a unique name was created,  the  last six  bytes  of template will have
+ *   been modified in such a way that the resulting name is unique
+ *   (i.e., does not exist already) If a unique name could not be created,
+ *   template is made an empty string, and errno is set to indicate the error.
  *
  ****************************************************************************/
 
-int mktemp(FAR char *path_template)
+FAR char *mktemp(FAR char *path_template)
 {
-	int fd = mkstemp(path_template);
-	if (fd < 0) {
-		return ERROR;
+	size_t len = strlen(path_template);
+	int retries = 100;
```

Comment:
I think earlier implementation has a large number of retries while finding the unique file name?